externs/shaka/manifest.js

```diff
@@ -107,6 +107,7 @@ shaka.extern.InitDataOverride;
  *   audioRobustness: string,
  *   videoRobustness: string,
  *   serverCertificate: Uint8Array,
+ *   sessionTypes: Array.<string>,
```

Comment:
Doing so would prevent to have specific sessionTypes for different keySystems.
In my case, I need to force the sessionType for PlayReady to `persistent-license`, but don't need it for Widevine. Setting it globally will affect each keySystems, I am not sure we want to do that 🤔

Reply:
DrmEngine is only used for a single key-system at a time.  So you can set the field after selecting the key system.

Reply:
@TheModMaker If you have any insight to share about this ?

Reply:
First, near 740, you'd use `info.keySystem` to get the advanced config to determine which session type to query for.  Then, when setting `this.currentDrmInfo_` near 900, you'd initialize the field based on which key system was selected.  When creating new sessions, you'd use the field directly.

Reply:
Thank you @TheModMaker :)

If I do so l.740, wouldn't it by-pass the work of `fillInDrmInfoDefaults_` done l.678 (with all the valuable checks) ? I would need to re-do some logic to get advanced configuration, check it exists, etc. I've added the `sessionTypes` in the type def, because I've done the exact same logic inside `fillInDrmInfoDefaults_`.

I could then init the `this.newSessionType_` just below each `requestMediaKeySystemAccess`, to use it then to create the session rather than referring to the current DrmInfo, which holds all the valuable information about the selected keySystem config.

For review's clarity, I've drafted the following diff, https://github.com/valotvince/shaka-player/compare/feat-playready-recommendation...valotvince:feat-playready-recommendation-diff, please let me know if this what you have in mind !

Thanks :)

Reply:
👋  @TheModMaker, any comments about my previous message ? Thank you ! 😄